### PR TITLE
Checking validity of Project Name

### PR DESCRIPTION
**Pull request by sparshypop7**

### CODE REVIEW

File: src/views/ProjectsDash.tsx [hunk 1]
```diff
@@ -46,8 +46,36 @@ export default function ProjectsDash() {
...
+          const msg = `Project name is too long. Please enter a name with ${pid.length-64} less characters.`;
+          Swal.showValidationMessage(
+            msg
+          );
+        } else if (name.length < 3) {
+          Swal.showValidationMessage(
+            "Please enter a valid project name. Project name cannot be less than 3 characters."
+          );
+        } else {
+          for(let i=0; i<arr_phrases.length; i++) {
```
Comment (techster10173) [inline src/views/ProjectsDash.tsx:71]: See if you want to replace the for-loop with this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes or if you want index try using `indexOf()`

File: src/views/ProjectsDash.tsx [hunk 2]
```diff
@@ -46,8 +46,36 @@ export default function ProjectsDash() {
...
+
+        if (!name) {
+          Swal.showValidationMessage(
+            "Please enter a project name"
+          );
+        } else if (!nameRegex.test(name)) {
+          Swal.showValidationMessage(
+            "Please enter a valid project name. Only lowercase letters and numbers are allowed."
+          );
+        } else if (pid.length > 64) {
```
Comment (techster10173) [inline src/views/ProjectsDash.tsx:61]: Change length tolerance to include sub accounts for hosting/storage

File: src/views/ProjectsDash.tsx [hunk 3]
```diff
@@ -46,8 +46,36 @@ export default function ProjectsDash() {
 
         const pid = `${name}-${uuid}.${window.accountId}`;
 
-        if (!nameRegex.test(name) || pid.length > 50) {
-          Swal.showValidationMessage("Invalid Project Name");
+        const name_length = pid.length-uuid.length-window.accountId.length-1;
+
+        const arr_phrases = ["near", "wallet", "account", "contract", "app", "test", "dev", "mainnet", "testnet", "helper"];
```
Comment (techster10173) [inline src/views/ProjectsDash.tsx:51]: Change variable name to INVALID_PHRASES or smth along those lines

**Review by techster10173 — CHANGES_REQUESTED** [review]
Make sure you merge main into this

**Review by techster10173 — APPROVED** [review]
LGTM!

**Review by srawulwar — APPROVED** [review]
lgtm

**Pull request merged**